tests/unit/buttons.js

```diff
@@ -38,6 +40,98 @@ define('tests/unit/buttons',
+            }
+        ));
+    });
+
+    describe('install button', function() {
+
+        it('is enabled for add-ons depending on device_type', h.injector(
+                h.mockSettings({
+                    addonsEnabled: true,
+                    model_prototypes: {app: 'slug'},
+                })
+            ).run(['buttons'], function(buttons) {
+                var product = {
+                    // Tests are running on a desktop browser.
+                    device_types: ['desktop'],
+                    // For now transform() needs a mini_manifest_url to
+                    // properly detect add-ons as such.
+                    mini_manifest_url : '/extension/lol.webapp',
+                };
+                result = buttons.transform(product);
+                assert.equal(result.disabled, undefined);
+                assert.equal(result.isAddon, true);
+                assert.equal(result.isLangpack, false);
+                assert.equal(result.incompatible, undefined);
+                assert.equal(result.installed, false);
+            }
+        ));
+
+        it('is disabled for add-ons depending on device_type', h.injector(
+                h.mockSettings({
+                    addonsEnabled: true,
+                    model_prototypes: {app: 'slug'},
+                })
+            ).run(['buttons'], function(buttons) {
+                var product = {
+                    // Tests are running on a desktop browser.
+                    device_types: ['foooooobar'],
+                    // For now transform() needs a mini_manifest_url to
+                    // properly detect add-ons as such.
+                    mini_manifest_url : '/extension/lol.webapp',
+                };
+                result = buttons.transform(product);
+                assert.deepEqual(result.disabled,
+                                 ['not available for your platform']);
+                assert.equal(result.isAddon, true);
+                assert.equal(result.isLangpack, false);
+                assert.deepEqual(result.incompatible,
+                                 ['not available for your platform']);
+                assert.equal(result.installed, false);
+            }
+        ));
+
+        it('is disabled for add-ons if setting is set to false', h.injector(
+                h.mockSettings({
+                    addonsEnabled: false,
+                    model_prototypes: {app: 'slug'},
+                })
+            ).run(['buttons'], function(buttons) {
+                var product = {
+                    // Tests are running on a desktop browser.
+                    device_types: ['desktop'],
+                    // For now transform() needs a mini_manifest_url to
+                    // properly detect add-ons as such.
+                    mini_manifest_url : '/extension/lol.webapp',
+                };
+                result = buttons.transform(product);
+                assert.deepEqual(result.disabled,
```

Comment:
Ah, yeah, good call. `getBtnText()` test above works because it calls `apps.transform()`, but weirdly `buttons.transform()` doesn't. I'll see if I can refactor things a bit to make it easier to follow.